gatsby/content/docs/e2ee_cross_signing.mdx

```diff
@@ -0,0 +1,920 @@
+For that, there are entries `m.secret_storage.key.<keyId>`, where the keyId is the unique ID of said
+key. In addition, there is an `m.secret_storage.default_key` which denotes the default key to use.
+
+The `m.secret_storage.<keyId>` objects contain a mac and an iv to easily verify if you have the correct
+key, and, in case if there is also a password associated with it, some information on how to generate
+the key out of the password. More specific on that later.
+
+Secrets, such as the private keys for cross-signing, can only be strings (not binary data) and are
+stored encrypted in the account data of your account. For example, the `m.cross_signing.master` entry
+in account data is the private key of your master cross-signing key. Each secret can be encrypted by
+multiple SSS keys. Inside the `encrypted` object there is a mapping of key ID to the ciphertext, iv
+and mac of the secret. A content could look as following:
+
+```json
+{
+  "encrypted": {
+    "awesomeKeyId": {
+      "iv": "supercooliv",
+      "ciphertext": "secretciphertext",
+      "mac": "macthingy"
+    }
+  }
+}
+```
+
+### Caching of secrets
+You have to access some secrets comparatively often. As fetching from the store requires to get a key
+for said secret, that requires user interaction. As such, if a client can store a secret seasonably
+securely client-sided, it is recommended to cache some secrets locally. For cross-signing that would
+be `m.cross_signing.self_signing`, `m.cross_signing.user_signing` and for online key backup that would
+be `m.megolm_backup.v1`. Blindly caching all secrets is a security risk, as some secrets such as
+`m.cross_signing.master` are highly sensitive and shouldn't be stored in plaintext on any disk, ever.
+
+You should always keep a whitelist of what secrets to cache, never a blacklist, as in the future some
+new highly-sensitive secret may be added.
+
+As a secret might change in the future, it might be a good idea to store in the secret cache some more
+information along with the secret - perhaps its ciphertext with key id? So that if either key id or
+ciphertext changes, you know that you'll have to fetch the secret from the store again, and your cache
+is out-of-date.
+
+### Sharing
+The same limitations of caching apply to sharing: It only makes sense to request secrets or share
+secrets that are cacheable, as else potentially sensitive data is leaked, or it requires user interaction
+to load those secrets in the first place anyways.
+
+Secrets can only be requested from other devices of your account that you already have verified, be
+it via cross-signing or directly. As such, it might be a good idea to request all cacheable secrets
+after successfully completing verification with yourself, as far as that also verifies your own master
+key (else the other client likely doesn't have cross signing and thus requesting secrets doesn't make
+much sense).
```

Comment:
If requesting secrets after completing verification, it may be worth introducing a small delay before requesting secrets, to make sure that the other client has had a chance to mark you as being trusted.

Reply:
Good point! *adds that to own implementation*